Sources/Rendering/OpenGL/glsl/vtkVolumeFS.glsl

```diff
@@ -951,15 +953,16 @@ vec2 computeRayDistances(vec3 rayDir, vec3 tdims)
 void computeIndexSpaceValues(out vec3 pos, out vec3 endPos, out float sampleDistanceIS, vec3 rayDir, vec2 dists)
 {
   // compute starting and ending values in volume space
-  pos = vertexVCVSOutput + dists.x*rayDir;
+  vec3 halfVoxel = vOriginPlusHalfVoxelVC - vOriginVC;
+  pos = vertexVCVSOutput + dists.x*rayDir + halfVoxel;
   pos = pos - vOriginVC;
```

Comment:
The code here looks odd. The ray position start and end points (in View Coords) are being offset by half a voxel. That seems confusing as the ray positions should be unchanged, it should be the mapping of the ViewCoords to volume TextureCoords that should be updated to me. Otherwise intersections with cropping planes etc get a bit off as the ray is no longer the original ray, but this offset ray. Does that make sense?

Reply:
Instead of correcting for the padding directly inside the shader, the spatial extents can be correctly set to pad by 0.5 voxels, like here:
https://github.com/Kitware/vtk-js/pull/2089/files#diff-653a0c06ccb1ec727a15802ff1d90b9cb6599f82d58610b2522dcfe483ab7cf1R168

This way, there would be minimal changes to the shader, like here:
https://github.com/Kitware/vtk-js/pull/2089/files#diff-4755e93326d6496e13ee38881a618a5accbe127aa1c4de56cdac22555b5fdc4cR909

Reply:
@jadh4v Thanks. As I see your PR has both image and volume mapper fixes but it is in Draft, can I ask what is the plan for your PR to get merged in master? We are in a rush to have this bug fixed for volumeMapper in vtk, but was wondering if yours is ready? if not I can go ahead adding the `getSpatialExtent` approach instead of modifying shader code.

Reply:
@sedghi We kinda got stuck into a discussion regarding how to handle medical vs simulation data (which doesn't require the padding), and convention regarding the definition of the `Origin` point given the padding.

But my PR already addresses the core issue of proper padding for both Image and Volume mappers. So I think we can quickly merge it. How about this, I will rebase and re-push the updated branch and share it with you. You can test it locally if it works for your use-case, and also push the unit test that you have created to the branch.

Reply:
@jadh4v Yes, that would be great! thanks for your help

Reply:
@sedghi  Great! I have updated my branch and added you as a collaborator. Please go ahead with further modifications: https://github.com/jadh4v/vtk-js/tree/image-volume-half-voxel

Reply:
There are some tests failing, but I can take care of it after you push your changes.

Reply:
@jadh4v Thanks again for your help, I tested your fixes, and it made sense and also works for our usecase. I added two commits in your fork:
1. Added my test for the volumeMapperBounds, but there are some tests failing as you mentioned, but the new added test is passing. 
2. Fixed the halfVoxel for the labelmapOutline